### ISSUE TITLE
`flux_bootstrap_git` cannot support >1 `GitRepository`

### ISSUE THREAD

**Issue by jamiezieziula** [fluxcd/terraform-provider-flux#429]
I did a deep dive into migrating from the old way of deploying flux to a GKE cluster to the new `flux_bootstrap_git` resource today and was unable to complete the migration due to two main issues with its current implementation:

1 - `flux_bootstrapt_git` cannot be used to deploy >1 `GitRepository` into a single cluster, within a single terraform workspace (directory).
- For some of our GKE clusters, we deploy 4 `GitRepository`s. With the old TF implementation, I just added some minor logic to conditionally run the `install` portion of the flux implementation - that way I could install flux components once and only once. With this new setup - there is no way to conditionally install the flux core components (CRDs).  I attempted to pass an empty array to the `components` parameter - but that resulted in the following error:
 ```
resource "flux_bootstrap_git" "repo_2" {
  path = "path/to/cluster/config"
  components = []
}
 ```
```
Error: Invalid Attribute Value
with flux_bootstrap_git.repo_2
on main.tf line 35, in resource "flux_bootstrap_git" "repo_2":
  components = []
Attribute components set must contain at least 2 elements, got: 0

Error: Missing Required Set Contents
with flux_bootstrap_git.repo_2
on main.tf line 35, in resource "flux_bootstrap_git" "repo_2":
  components = []
Set has to contain the items [source-controller kustomize-controller]
```

2 - there is no parameter in the `flux_bootstrap_resource` that allows you to override the name of the base `GitRepository` and `Kustomization`. The default (and unchangeable) value for these resources is `flux-system`. 
- This is equally relevant in allowing a user to implement >1 `GitRepository` in a single cluster because the `GitRepository` and `Kustomization` name cannot be changed, and therefore would conflict with each other.

I would venture to guess that at least _some_ of fluxcd users are running more than a single `GitRepository` in their clusters, so I think that both of these configuration options would be relevant and worthwhile to implement.

**Comment by JulesLalu** [comment]
Hello, I am facing the second issue as well. Could we have some updates on this, or at least a workaround ?

**Comment by stefanprodan** [comment]
> I would venture to guess that at least some of fluxcd users are running more than a single GitRepository in their clusters, so I think that both of these configuration options would be relevant and worthwhile to implement.

I expect users to create extra GitRepositories and Kustomizations directly in the bootstrap repo instead of applying them directly on the cluster with Terraform. Using TF for this breaks the GitOps promise, as any update to these objects need cluster access instead of doing them directly in the bootstrap repo. Anyway, if you really want to do this, please see #480

**Comment by JulesLalu** [comment]
Thank you for opening this PR, which solves the first issue and could be used as a workaround for the second one, too. However, would it be possible to open another PR to enable overriding the name of the base `GitRepository` and `Kustomization` in `flux_bootstrap_resource` ? I feel like this would be a lot easier to use in case of using one single repo, but that is not named 'flux-system'.

**Comment by stefanprodan** [comment]
> However, would it be possible to open another PR to enable overriding the name of the base GitRepository and Kustomization in flux_bootstrap_resource ?

I'm not for allowing this, the Terraform provider must subscribe to Flux bootstrap conventions as implemented in the CLI.
